spec/components/candidate_interface/previous_applications_component_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::PreviousApplicationsComponent do
+  describe 'a current application ' do
+    context 'with an ACCEPTED state' do
+      let(:current_application_form) { create_application_form_with_course_choices(statuses: [status]) }
+
+      ApplicationStateChange::ACCEPTED_STATES.each do |status|
+        let(:status) { status }
+
+        it "does not render the component for accepted state: '#{status}'" do
+          result = render_inline(described_class.new(application_form: current_application_form))
+
+          expect(result.css('.app-table__caption--m').text).to be_empty
+        end
+      end
+    end
+
+    context 'without an ACCEPTED state' do
+      let(:current_application_form) { create_application_form_with_course_choices(statuses: %w[rejected]) }
+
+      it 'renders component with unsuccessful application choice in a table' do
+        unsuccessful_application_choice = current_application_form.application_choices.first
+        result = render_inline(described_class.new(application_form: current_application_form))
+
+        expect(result.css('.app-table__caption--m').text).to include('Your previous applications')
+        expect(result.css('.app-course-choice__provider-name').text).to include(unsuccessful_application_choice.course.provider.name)
+        expect(result.css('.app-course-choice__course-name').text).to include("#{unsuccessful_application_choice.course.name} (#{unsuccessful_application_choice.course.code})")
+        expect(result.css('.app-tag').text).to include('Unsuccessful')
+        expect(result.css('.govuk-link').text).to include('View application')
+      end
+    end
+  end
+
+  describe 'a previous application exists with a rejected application choice' do
+    let(:current_application_form) { create(:completed_application_form, submitted_at: 3.days.ago, previous_application_form: previous_application_form) }
+    let(:previous_application_form) { create(:completed_application_form, submitted_at: 10.days.ago) }
+    let!(:unsuccessful_application_choice) { create(:application_choice, :with_rejection, application_form: previous_application_form) }
+
+    it 'renders component with unsuccessful application choice in a table' do
+      result = render_inline(described_class.new(application_form: current_application_form))
+
+      expect(result.css('.app-table__caption--m').text).to include('Your previous applications')
+      expect(result.css('.app-course-choice__provider-name').text).to include(unsuccessful_application_choice.course.provider.name)
+      expect(result.css('.app-course-choice__course-name').text).to include("#{unsuccessful_application_choice.course.name} (#{unsuccessful_application_choice.course.code})")
+      expect(result.css('.app-tag').text).to include('Unsuccessful')
+      expect(result.css('.govuk-link').text).to include('View application')
+    end
+  end
+
+  describe '#application_choices' do
```

Comment:
Same here re: https://github.com/DFE-Digital/apply-for-teacher-training/pull/4368/files#r601568795

Reply:
As above this makes sense to me. I guess we could test that page doesn't include 'Your previous applications', but i'm not convinced it's necessary